index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2019-present, ProductBoard, Inc.
+ * All rights reserved.
+ */
+
+const fs = require("fs");
+const child = require("child_process");
+const gzipSize = require("gzip-size");
+
+/**
+ * Webpack Plugin for generating info JSON (hash, sha, assetSize), it's consumable by webpack-deploy later on when deploying
+ */
+class BuildInfo {
+  constructor({ filename = "build.log" }) {
+    this.filename = filename;
+  }
+
+  getSha() {
+    return child
+      .execSync("git rev-parse HEAD")
+      .toString()
+      .trim()
+      .slice(0, 7);
+  }
+
+  async createBuildInfoReport(compilation) {
+    const hash = compilation.getStats().toJson().hash;
+
+    const assetsSize = await Promise.all(
+      Object.entries(compilation.assets)
+        .filter(([asset]) => /.js$/.test(asset))
+        .map(async ([asset, content]) => {
+          const gzipedSize = await gzipSize(content.source());
+          return {
+            asset,
+            size: content.size(),
+            gzipedSize
+          };
+        })
+    );
+
+    // get SHA from git
+    const sha = this.getSha();
+
+    const info = JSON.stringify({
+      sha,
+      hash,
+      assetsSize
+    });
+
+    fs.writeFileSync(this.filename, info);
```

Comment:
Doesn't it requite any updates on the consuming side?

Reply:
There is callback in `apply` if you are asking about if it communicate correctly to the Webpack